plugins/junit/src/main/java/org/apache/struts2/StrutsJUnit4TestCase.java

```diff
@@ -170,6 +168,20 @@ protected ActionProxy getActionProxy(String uri) {
         return proxy;
     }
 
+    private void initActionContext(ActionContext actionContext) {
+        actionContext.setParameters(HttpParameters.create(request.getParameterMap()).build());
+        initSession(actionContext);
+        // set the action context to the one used by the proxy
+        ActionContext.setContext(actionContext);
+    }
+
+    private void initSession(ActionContext actionContext) {

```

Comment:
Same here, making methods protected allows users to override them.

Reply:
Ah... again I forgot this practice. Those are IntelliJ auto created methods. Thanks! fixed :)